doc/BUILD.md

```diff
@@ -153,6 +153,18 @@ colcon build --symlink-install
 In the case that the developer wants to change any dependencies, they can run
 `<directory_for_workspaces>/navigation2_ws/src/navigation2/tools/build_all.sh` in a clean environment to get everything rebuilt easily
 
+### Debugging
+
+To build Navigation2 with build symbols, use colcon build with the following flags and cmake arguments.
```

Comment:
Unless something very odd is happening in colcon, that's not compiling with symbols. You need to actually combine with gdb flags, `-g` in the CMake settings. That just does a debug build.

Reply:
Setting `CMAKE_BUILD_TYPE=Debug` automatically adds the `-g` flag and also disables optimization. Nothing to do with colcon. That's how CMake works by default.

Reply:
_really_ 

**what have I been doing all these years**

Reply:
I think a lot of builds add CMake instructions to manually mess with the `-g` flag, so if you start from a build system that does that, the `CMAKE_BUILD_TYPE` flag wouldn't work as intended, leading you to believe that it doesn't just "do the right thing"

There's also a RelWithDebInfo build type that adds the -g flag and sets -O2 optimization. I've been wondering if we should build that way by default, so if anybody ever gives us a core dump, we can see what's going on.

Reply:
`-g` shouldn't go in the build farm, if that's what you're proposing. You mean just for local builds?

Reply:
No. I'm saying the build farm could/(should?) build with -g, then strip the binaries to remove the symbols, but save them off. No symbols would be present in the release. They would just be saved off for future reference if needed.

Without that, you can not debug a released build. If the symbols for the release were saved off, people running into a problem could post a core dump, we could open that up in a debugger and see what's going on.